service/qos/service_level_controller.cc

```diff
@@ -333,9 +339,7 @@ future<std::optional<service_level_options>> service_level_controller::find_effe
                     return std::nullopt;
                 }
 
-                if (include_names == include_effective_names::yes) {
-                    sl_it->second.slo.init_effective_names(*sl_name);
-                }
+                sl_it->second.slo.init_effective_names(*sl_name);
```

Comment:
What's the reason for this change? I don't understand why we were doing it in the first place, but I also don't understand why it's OK to remove the `include_names` parameter and set the effective SL names unconditionally.

Reply:
Effective names were added later to support `LIST EFFECTIVE SERVICE LEVEL OF <role>`. It wasn't needed anywhere else, so I made it optional to include the names.
But in effective service levels cache the effective names are always included, so also made it unconditional in old path (without effective sl cache) in `find_effective_service_level()`